From b97c97cd3b079242a3d7e3855257b7a18641fe84 Mon Sep 17 00:00:00 2001
From: Eike Ziller <eike.ziller@theqtcompany.com>
Date: Wed, 17 Jun 2015 15:33:44 +0200
Subject: [PATCH] Disable Spotlight locator filter on OS X SDK < 10.10

One of the used constants only exists since 10.9, and it seems that
compilation on 10.9 can create broken setups as well.

Change-Id: Icddedb77c77efca82ca00090c8a5d53db3284c96
Task-number: QTCREATORBUG-14525
---
 src/plugins/coreplugin/locator/locator.cpp               | 4 ++--
 src/plugins/coreplugin/locator/spotlightlocatorfilter.h  | 6 ++++++
 src/plugins/coreplugin/locator/spotlightlocatorfilter.mm | 6 ++++++
 3 files changed, 14 insertions(+), 2 deletions(-)

diff --git src/plugins/coreplugin/locator/locator.cpp src/plugins/coreplugin/locator/locator.cpp
index 095ffa5..758a8e3 100644
--- src/plugins/coreplugin/locator/locator.cpp
+++ src/plugins/coreplugin/locator/locator.cpp
@@ -59,7 +59,7 @@
 #include <QFuture>
 #include <QAction>
 
-#ifdef Q_OS_OSX
+#if defined(Q_OS_OSX) && QT_MAC_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_10, __IPHONE_NA)
 #include "spotlightlocatorfilter.h"
 #endif
 
@@ -129,7 +129,7 @@ void Locator::initialize(CorePlugin *corePlugin, const QStringList &, QString *)
     m_corePlugin->addObject(m_externalToolsFilter);
 
     m_corePlugin->addAutoReleasedObject(new LocatorFiltersFilter(this, m_locatorWidget));
-#ifdef Q_OS_OSX
+#if defined(Q_OS_OSX) && QT_MAC_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_10, __IPHONE_NA)
     m_corePlugin->addAutoReleasedObject(new SpotlightLocatorFilter);
 #endif
 
diff --git src/plugins/coreplugin/locator/spotlightlocatorfilter.h src/plugins/coreplugin/locator/spotlightlocatorfilter.h
index 41ecef9..c208e99 100644
--- src/plugins/coreplugin/locator/spotlightlocatorfilter.h
+++ src/plugins/coreplugin/locator/spotlightlocatorfilter.h
@@ -32,6 +32,10 @@
 #ifndef SPOTLIGHTLOCATORFILTER_H
 #define SPOTLIGHTLOCATORFILTER_H
 
+#include <qglobal.h>
+
+#if QT_MAC_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_10, __IPHONE_NA)
+
 #include "basefilefilter.h"
 
 namespace Core {
@@ -49,5 +53,7 @@ public:
 } // Internal
 } // Core
 
+#endif // QT_MAC_PLATFORM_SDK_EQUAL_OR_ABOVE
+
 #endif // SPOTLIGHTLOCATORFILTER_H
 
diff --git src/plugins/coreplugin/locator/spotlightlocatorfilter.mm src/plugins/coreplugin/locator/spotlightlocatorfilter.mm
index 4b63274..8d452b8 100644
--- src/plugins/coreplugin/locator/spotlightlocatorfilter.mm
+++ src/plugins/coreplugin/locator/spotlightlocatorfilter.mm
@@ -28,6 +28,10 @@
 **
 ****************************************************************************/
 
+#include <qglobal.h>
+
+#if QT_MAC_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_10, __IPHONE_NA)
+
 #include "spotlightlocatorfilter.h"
 
 #include <utils/autoreleasepool.h>
@@ -215,3 +219,5 @@ void SpotlightLocatorFilter::refresh(QFutureInterface<void> &future)
 
 } // Internal
 } // Core
+
+#endif // QT_MAC_PLATFORM_SDK_EQUAL_OR_ABOVE
-- 
2.4.3

